src/Stripe.elm

```diff
@@ -121,52 +121,56 @@ decodeCurrency =
         D.string
 
 
+type CheckoutItem
+    = Priced
+        { priceId : Id PriceId
+        , name : String
+        , quantity : Int
+        }
+    | Unpriced
+        { name : String
+        , quantity : Int
+        , currency : String
+        , amountDecimal : Int
+        }
+
+
 createCheckoutSession :
-    { priceId : Id PriceId
-    , opportunityGrantDonation : Int
+    { items : List CheckoutItem
     , emailAddress : EmailAddress
-    , sponsorship : Maybe String
     , now : Time.Posix
     , expiresInMinutes : Int
     }
     -> Task Http.Error (Id StripeSessionId)
-createCheckoutSession { priceId, opportunityGrantDonation, emailAddress, sponsorship, now, expiresInMinutes } =
-    -- @TODO support multiple prices, see Data.Tickets
+createCheckoutSession { items, emailAddress, now, expiresInMinutes } =
     let
-        opportunityGrantAttrs =
-            if opportunityGrantDonation > 0 then
-                [ ( "line_items[1][price_data][currency]", "eur" )
-                , ( "line_items[1][price_data][product_data][name]", "Elm Camp Denmark 23 - Opportunity Grant Sponsorship" )
-                , ( "line_items[1][price_data][product_data][description]", "Thank you for your generous donation!" )
-                , ( "line_items[1][price_data][unit_amount_decimal]", String.fromInt (opportunityGrantDonation * 100) )
-                , ( "line_items[1][quantity]", "1" )
-                ]
-
-            else
-                []
-
-        sponsorIndex =
-            if opportunityGrantAttrs == [] then
-                "1"
-
-            else
-                "2"
-
-        sponsorshipAttrs =
-            case sponsorship of
-                Just s ->
-                    [ ( "line_items[" ++ sponsorIndex ++ "][price]", s )
-                    , ( "line_items[" ++ sponsorIndex ++ "][quantity]", "1" )
+        -- opportunityGrantAttrs =
```

Comment:
Does this need putting back in?

Reply:
Ah nope, was left for reference while I was converting the Stripe implementation to be completely generic 👍